ui/src/services/PodUtils.js

```diff
@@ -0,0 +1,52 @@
+import { fromMilliSectoAge } from './utils.js';
+
+// Return the pod list for the Pod Tab in Node Page
+export const getPodsListData = (nodeName, pods) => {
+  const podsList = pods?.filter((pod) => pod.nodeName === nodeName);
+
+  return podsList?.map((pod) => {
+    let ageObject, age;
+    ageObject = fromMilliSectoAge(new Date() - pod.startTime);
+    // There are in total 10 cases to display the `Age` in two time unit maximum.
+    if (ageObject.day > 0 && ageObject.hour > 0) {
+      // start with `d`
+      age = ageObject.day + 'd' + ageObject.hour + 'h';
+    } else if (ageObject.day > 0 && ageObject.minute > 0) {
+      age = ageObject.day + 'd' + ageObject.minute + 'm';
+    } else if (ageObject.day > 0 && ageObject.second > 0) {
+      age = ageObject.day + 'd' + ageObject.second + 's';
+    } else if (ageObject.day > 0) {
+      age = ageObject.day + 'd';
+    } else if (ageObject.hour > 0 && ageObject.minute > 0) {
+      // start with `h`
+      age = ageObject.hour + 'd' + ageObject.minute + 'm';
+    } else if (ageObject.hour > 0 && ageObject.second > 0) {
+      age = ageObject.hour + 'd' + ageObject.second + 's';
+    } else if (ageObject.hour > 0) {
+      age = ageObject.hour + 'h';
+    } else if (ageObject.mintue > 0 && ageObject.second > 0) {
+      // start with `m`
+      age = ageObject.minute + 'm' + ageObject.second + 's';
+    } else if (ageObject.mintue > 0) {
+      age = ageObject.minute + 'm';
+    } else if (ageObject.second > 0) {
+      // start with `s`
+      age = ageObject.second + 's';
+    }
```

Comment:
I'm confident this logic should be extract and unit tested.
I'm not sure if this can be easily "simplify" so even though it does not look very smart. It's easily understandable so let's keep it that way.
Also maybe provide some example will help to quickly understand the purpose.

Reply:
+1 for the unit test, that would catch:
- the typo (`s/mintue/minute/`) l.27 and l. 30
- the copy/paste error (`d` should be `h` l.22 and l.24)

And maybe other errors.

What if all fields are zero (unlikely but not impossible)? What will you return?

As for simplification: you could write that:

```javascript
  const age = [];
  
  if (ageObject.day > 0) {
    age.push(ageObject.day + 'd');
  }
  if (ageObject.hour > 0) {
    age.push(ageObject.hour + 'h');
  }
  if (ageObject.minute > 0) {
    age.push(ageObject.minute + 'm');
  }
  if (ageObject.second > 0) {
    age.push(ageObject.second + 's');
  }
  return age.slice(0, 2).join('');
```
=> Add every non-zero value and only take the two first.

Reply:
Thanks for this smart and neat solution!